### PR TITLE
Allow commenting on submit request diff lines

### PR DESCRIPTION
**Pull request by hellcp-work**
Depends on #13910 

Allows users to comment on submit request action diffs.

Screenshot from 2023-02-28 13-06-33

# How to test

1. Create a submit request with a diff
2. Open changes tab
3. Hover over a line
4. Click a button that shows up
5. Write a comment and save it

### CODE REVIEW

File: src/api/app/views/webui/request/inline_comment.js.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+$('#flash').empty();
+$('#comment<%= @line %> .comments-list').animate({
+  opacity: 0.25
+}, 400, function() {
+  $('#comment<%= @line %> .comments-list').html("<%= escape_javascript(render(partial: 'webui/request/inline_comment', locals: { comment: @comment, commentable: @active_action, diff_ref: @line })) %>");
+  $('#comment<%= @line %> .comments-list').animate({ opacity: 1 }, 400, function() {
```
Comment (danidoni) [inline src/api/app/views/webui/request/inline_comment.js.erb:6]: I believe this feature would be even better without the fade in/fade out

File: src/api/spec/components/diff_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  let(:target_package) { create(:package_with_file, name: 'target_package', project: target_project, file_name: file_name, file_content: '# This will be replaced') }
+  let(:source_package) { create(:package_with_file, name: 'source_package', project: source_project, file_name: file_name, file_content: '# This is the new text') }
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           target_package: target_package,
+           source_package: source_package)
+  end
+  let(:bs_request_opts) { { filelimit: nil, tarlimit: nil, diff_to_superseded: nil, diffs: true, cacheonly: 1 } }
+
+  context 'with a request with a submit action' do
```
Comment (danidoni) [inline src/api/spec/components/diff_component_spec.rb:16]:
```suggestion
  context 'with a request having a submit action' do
```

Comment (danidoni) [reply src/api/spec/components/diff_component_spec.rb:16]: Sounds nicer without repeating `with` :thought_balloon:

**Comment by rubhanazeem** [comment]
The comment section stays open when you add a new comment. Is it possible to remove that additional comment section?

File: src/api/app/models/comment.rb [hunk 3]
```diff
@@ -1,14 +1,15 @@
 require 'set'
 
 class Comment < ApplicationRecord
-  belongs_to :commentable, polymorphic: true # belongs to a Project, Package or BsRequest
+  belongs_to :commentable, polymorphic: true # belongs to a Project, Package, BsRequest or BsRequestActionSubmit
```
Comment (rubhanazeem) [inline src/api/app/models/comment.rb:4]: We can change `BsRequestActionSubmit` to `BsRequestAction`

Comment (rubhanazeem) [reply src/api/app/models/comment.rb:4]: It's correct. I was referring to the association itself. Just checked that the code is specifically checking `BsRequestActionSubmit` :+1:

**Review by danidoni — APPROVED** [review]

**Review by rubhanazeem — APPROVED** [review]

**Comment by DimStar77** [comment]
Tried this new feature:

https://build.opensuse.org/request/show/1069679

I doubt that people will 'find' the comments and act on them

**Comment by hennevogel** [comment]
@DimStar77 integrating this into conversation etc. is in the backlog https://trello.com/c/X62LRP8N

**Pull request merged**